src/utils.rs

```diff
@@ -152,6 +152,66 @@ pub fn ensure_procfs(path: &Path) -> Result<()> {
     Ok(())
 }
 
+pub fn secure_join(rootfs: &Path, unsafe_path: &Path) -> Result<PathBuf> {
+    
+    let mut clean_path = PathBuf::from(rootfs);
+    let mut path = PathBuf::from(unsafe_path);
+    let mut p = PathBuf::new();
+
+    let mut part = path.iter();
+    let mut i = 0;
+
+    loop {
+        if i > 255 {
+            // Dereference too many symlinks, may be infinite loop
```

Comment:
Can we error out in this use? You can use `bail!("dereference too many symlinks, may be infinite loop");`

Reply:
You can remove this comment now.